test/cmdlineTests/optimize_full_storage_write/output

```diff
@@ -42,6 +42,9 @@ sub_0: assembly {
       eq
       tag_3
       jumpi
+      0x00
+      dup1
+      revert
```

Comment:
So one thing is that this is no longer de-duplicated with ``tag_2`` below for some reason (may be that the ``0x00`` is a ``PUSH0`` in one case and a ``PUSH1 0x00`` in the other?). Still a bit strange how the changes in this PR are causing this.

But another thing is: I'd have assumed that the libevmasm/CommonSubexpressionEliminator should actually be clever enough to transform ``0x00 dup1 revert`` to ``0x00 0x00 revert``, if it gets the correct push0 pricing.

Reply:
So, here's what happens in this case. The `inliner` sees that the cost of inlining is less than the `pushtag 2`, which was not the case before `push0`. So the code gets inlined. After that, the `peephole` optimizer finds that it can replace the code with `revert dup1 push0`, because its method `DoublePush` is not prepared to handle `push0`. Another problem is that the `peephole` optimizer was able to remove the `pushtag 2` after the `jump`, but it cannot do it now, because the inlining results in the block following the `jump` ending with a `revert`, which prevents the removal (see `UnreachableCode` method of `peephole`).
Finally, the CSE is not able to transform `dup1 push0` to `push0 push0`. I guess it needs a rule for that in `SimplificationRules`. 
Not sure about what to do regarding the `inliner` situation.
I will try to make the CSE and `peephole` optimizer consider `push0`, although for the latter that depends on getting the evm version in a static function of the optimizer (`DoublePush::applySimple`).

Reply:
Ah, right, that explains things - I hadn't considered that inlining would kick in since assuming a tag size of two bytes ``PUSH0 PUSH0 REVERT`` actually now is already the same size as ``PUSH<tag> JUMP``! I mean, letting the inlining happen is just fine then...

The peephole optimizer should definitely consider ``push0`` - and it also looks like some deduplication still doesn't happen correctly (but in cases in which it also didn't happen before), so we don't necessarily need to fix it here directly.